wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Field.kt

```diff
@@ -117,8 +117,13 @@ class Field private constructor(
     if (isPacked && !isPackable(linker, type!!)) {
       linker.addError("packed=true not permitted on $type")
     }
-    if (isExtension && isRequired) {
-      linker.addError("extension fields cannot be required")
+    if (isExtension) {
+      if (isRequired) {
+        linker.addError("extension fields cannot be required")
+      }
+      if (type!!.isMap) {
+        linker.addError("map fields cannot be extensions")
```

Comment:
```suggestion
        linker.addError("extension fields cannot be a map")
```
For consistency with above message

Reply:
fixed